lib/project.ts

```diff
@@ -423,8 +423,19 @@ export class Project extends ProjectBase implements Project.IProject {
 					_.each(this.configurationSpecificData, data => this.$projectPropertiesService.updateProjectProperty(data, undefined, mode, normalizedPropertyName, propertyValues).wait());
 				}
 			}
+		}).future<void>()();
+	}
+
+	public updateProjectPropertyAndSave(mode: string, propertyName: string, propertyValues: string[], configurations?: string[]): IFuture<void> {
+		return (() => {
+			this.ensureProject();
```

Comment:
You can extract the beginning of `updateProjectPropertyAndSave` and `updateProjectProperty` into a separate private method, which returns information about `projectConfigurations` and `normalizedPropertyName` in order to avoid code duplication